plan/plan.go

```diff
@@ -179,43 +179,30 @@ func shouldUpdateTTL(desired, current *endpoint.Endpoint) bool {
 }
 
 func shouldUpdateProviderSpecific(desired, current *endpoint.Endpoint) bool {
-	if current.ProviderSpecific == nil && len(desired.ProviderSpecific) == 0 {
-		return false
-	}
-	for _, c := range current.ProviderSpecific {
-		// don't consider target health when detecting changes
-		// see: https://github.com/kubernetes-sigs/external-dns/issues/869#issuecomment-458576954
-		if c.Name == "aws/evaluate-target-health" {
-			continue
-		}
+	desiredProperties := map[string]endpoint.ProviderSpecificProperty{}
 
-		found := false
+	if desired.ProviderSpecific != nil {
```

Comment:
It's not at all obvious, but I think the things in `desired.ProviderSpecific` probably don't have their `PropertyKind` set here (since they come from an annotation on an Ingress, and there's no code to set it there).

One of the things about the existing type is that it accurately reflects the information we've been given by the user (in their Ingress), whereas now there's this extra piece of information that isn't set.

Reply:
The Kind is used only from current records, so it's not an issue

Reply:
Yes, I can see that, but there's no comment to that effect in the code and it doesn't necessarily feel right making all the properties carry round extra information that isn't actually filled in.

As you have it here, if someone mistakenly changed it to use d.Kind, it would break. Or if they tried to use d.Kind somewhere else in the code, actually.

Reply:
Fundamentally the issue is that the `Kind` should really be associated with the property *name*, not with the property *value*. The problem is that unless you do what I did and add a `BaseProvider`, there's no obvious way to let the provider tell you what type things are without shoehorning it into the `ProviderSpecificProperty` struct.

Reply:
I see what you mean it makes sense